docs/development/doc_guide.rst

```diff
@@ -1,76 +1,849 @@
+* `IPython.sphinxext.ipython_console_highlighting`_
+* `sphinx_changelog`_ for rendering `towncrier`_ changelogs
+* `plasmapy_sphinx` for customizations created for use in PlasmaPy and
+  affiliated packages. Note that `plasmapy_sphinx` is expected to be
+  broken out into its own package in the future.
+
+These extensions are specified in :confval:`extensions` configuration
+value in `docs/conf.py`_.
+
+.. _external-references:
+
+References to other packages
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Intersphinx_ allows the automatic generation of links to the
+documentation of objects in other projects. This cross-package linking
+is made possible with the `sphinx.ext.intersphinx` extension.
+
+When we include ``astropy.units.Quantity`` in the documentation, it will
+show up as `astropy.units.Quantity` with a link to the appropriate page
+in Astropy documentation. Similarly, ``~astropy.units.Quantity`` will
+show up as `~astropy.units.Quantity`.
+
+The mappings are defined in the :confval:`intersphinx_mapping`
+dictionary in `docs/conf.py`_, and include Python_, NumPy_, SciPy_,
+Astropy_, pandas_, Sphinx_, and `sphinx_automodapi`_.
+
+New source packages may be added, but please verify that references to a
+function or class in that package show up correctly in `PlasmaPy's
+documentation`_. The name of the package does not always link as
+expected.
+
+.. hint::
+
+   If a cross-link is not working as expected this is usually due to one
+   of the following reasons:
+
+   * A typo;
+   * The package not being defined in :confval:`intersphinx_mapping`, or
+   * The referenced source package not properly or fully indexing their
+     own code, which is common in Python_ packages.
+
+Substitutions
+~~~~~~~~~~~~~
+
+Some functions and classes are referred to repeatedly throughout the
+documentation. reST_ allows us to `define substitutions
+<https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#substitution-definitions>`_.
+
+.. code-block:: rst
+
+   .. |Particle| replace:: `~plasmapy.particles.particle_class.Particle`
+
+PlasmaPy has certain common substitutions pre-defined so that they can
+be used elsewhere in the documentation. For example, we can write
+``|Quantity|`` instead of ``~astropy.units.Quantity``, and
+``|Particle|`` instead of
+``~plasmapy.particles.particle_class.Particle``.  For an up-to-date list
+of substitutions, please refer to the `docs/common_links.rst`_ file.
+
+Since substitutions are performed by Sphinx_ when the documentation is
+built, any substitution used in docstrings will not show up when using
+Python's `help` function (or the like). For example, when ``|Particle|``
+is used in a docstring, `help` will show it as ``|Particle|`` rather
+than ``~plasmapy.particles.particle_class.Particle``.  Consequently,
+substitutions should not be used in docstrings when it is important that
+users have quick access to the full path of the `object` (such as in the
+``See Also`` section).
```

Comment:
```suggestion
PlasmaPy has certain common substitutions pre-defined so that they can
be used elsewhere in the documentation. For example, we can write
``|Quantity|`` instead of ```~astropy.units.Quantity```, and
``|Particle|`` instead of
```~plasmapy.particles.particle_class.Particle```.  For an up-to-date list
of substitutions, please refer to the `docs/common_links.rst`_ file.

Since substitutions are performed by Sphinx_ when the documentation is
built, any substitution used in docstrings will not show up when using
Python's `help` function (or the like). For example, when ``|Particle|``
is used in a docstring, `help` will show it as ``|Particle|`` rather
than ```~plasmapy.particles.particle_class.Particle```.  Consequently,
substitutions should not be used in docstrings when it is important that
users have quick access to the full path of the `object` (such as in the
``See Also`` section).
```